t/concurrency_test.t

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/perl
+use strict;
+use warnings FATAL => 'all';
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use lib "$FindBin::Bin/../thirdparty/lib/perl5";
+use experimental 'signatures';
+use Test::More;
+use Time::HiRes qw (usleep);
+
+use Wireguard::WGmeta::Wrapper::ConfigT;
+use Wireguard::WGmeta::Utils;
+
+use constant TEST_DIR => $FindBin::Bin . '/test_data/';
+
+my $THREADS_PRESENT;
```

Comment:
Instead of using threads, which is probably not how this will be used IRL anyway, I would use fork instead ... 
or even better just open the file twice in the same code ...